tests/test_cdf_tk/test_auth.py

```diff
@@ -0,0 +1,203 @@
+import json
+import platform
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Union
+from unittest.mock import MagicMock
+
+import pytest
+import yaml
+from cognite.client._api.iam import IAMAPI
+from cognite.client.data_classes._base import CogniteResource, CogniteResourceList
+from cognite.client.data_classes.capabilities import (
+    AllProjectsScope,
+    ProjectCapability,
+    ProjectCapabilityList,
+)
+from cognite.client.data_classes.iam import Group, GroupList, ProjectSpec, TokenInspection
+from pytest import MonkeyPatch
+
+from cognite_toolkit.cdf_tk.bootstrap import check_auth
+from cognite_toolkit.cdf_tk.utils import CDFToolConfig
+from tests.conftest import ApprovalCogniteClient
+
+THIS_FOLDER = Path(__file__).resolve().parent
+
+TEST_PREFIX = "auth"
+DATA_FOLDER = THIS_FOLDER / f"{TEST_PREFIX}_data"
+SNAPSHOTS_DIR = THIS_FOLDER / f"{TEST_PREFIX}_data_snapshots"
+SNAPSHOTS_DIR.mkdir(exist_ok=True)
+
+
+@dataclass
+class CDFDataFixture:
+    cdf_tool: CDFToolConfig = None
+    approval_client: ApprovalCogniteClient = None
+    # The data dict is used to keep the pre-loaded CDF project data.
+    # Each CogniteResource needs to be added to the approval_client after the test
+    # has modified the data for its test purposes.
+    # E.g.
+    #
+    # approval_client.append(TokenInspection, modified_inspect_result)
+    # approval_client.append(Group, modified_group_list)
+    data: dict[CogniteResource, Union[CogniteResource, CogniteResourceList]] = None
+
+
+@pytest.fixture
+def cdf_data_fixture(
+    cognite_client_approval: ApprovalCogniteClient,
+    monkeypatch: MonkeyPatch,
+) -> CDFDataFixture:
```

Comment:
Instead of this one I would have two fixtures
`cdf_tool_config` and `cdf_resources`.

And considered extending the approval client with the changes you do to it, alternative make a `auth_verify_approval_client` fixture.

The advantage is that each of these fixtures becomes one responsible and easier to read and reuse.